packages/itwin/tree-widget/README.md

```diff
@@ -393,3 +393,129 @@ function MyWidget() {
   );
 }
 ```
+
+For custom tree components `TelemetryContextProvider` should be used:
+
+```tsx
+import { TelemetryContextProvider } from "@itwin/tree-widget-react";
+
+function MyWidget() {
+  return <TelemetryContextProvider
+    componentIdentifier="MyTree"
+    onPerformanceMeasured={(feature, elapsedTime) => {
+      console.log(`TreeWidget [${feature}] took ${elapsedTime} ms`)
+    }}
+    onFeatureUsed={(feature) => {
+      console.log(`TreeWidget [${feature}] used`)
+    }}
+  >
+    <MyTree />
+  </TelemetryContextProvider>;
+}
+
+function MyTree() {
+  // see "Custom trees" section for example implementation
+}
+```
+
+## Custom trees
+
+Some components are available for building custom trees that look and feel similar to the trees provided by this package:
+- `Tree` - base tree component that manages tree state, selection and filtering.
+- `TreeRenderer` - default renderer for tree data.
+- `VisibilityTree` - same as `Tree` component but additionally it can manage visibility of instances represented by tree nodes.
+- `VisibilityTreeRenderer` - same as `TreeRenderer` but additionally it renders checkboxes for visibility control.
```

Comment:
I think it would make sense to have two separate sections for these two types of components. For example:

```md
### Custom trees

The package delivers a set of building blocks for creating trees that look and feel similar to the tree components provided by this package. 

#### Custom basic tree

A "basic" tree is a tree that renders the hierarchy without visibility control - see [iModel content tree](#imodel-content-tree) for an example. Core components:
- `Tree` - component that manages tree state, selection and filtering.
- `TreeRenderer` - default renderer for tree data.

Example:
...

#### Custom visibility tree

A visibility tree is a tree that renders the hierarchy and allows controlling visibility control through the use of "eye" checkboxes - see [Models](#models-tree) and [Categories](#categories-tree) trees. Core components:
- `VisibilityTree` - same as `Tree` component but additionally manages visibility of instances represented by tree nodes.
- `VisibilityTreeRenderer` - same as `TreeRenderer` but additionally renders checkboxes for visibility control.

Example:
...

```